src/effects/Outline.tsx

```diff
@@ -1,25 +1,50 @@
 import { OutlineEffect } from 'postprocessing'
-import React, { Ref, ForwardRefExoticComponent, forwardRef, useMemo } from 'react'
+import React, { Ref, ForwardRefExoticComponent, forwardRef, useMemo, useLayoutEffect, useEffect } from 'react'
 import { useThree } from 'react-three-fiber'
+import { Object3D } from 'three'
 
-type OutlineProps = ConstructorParameters<typeof OutlineEffect>[2]
+type ObjectRef = Ref<Object3D>
+
+type OutlineProps = ConstructorParameters<typeof OutlineEffect>[2] | {
+  selection: ObjectRef | ObjectRef[],
+  selectionLayer: number
+}
 
 export const Outline: ForwardRefExoticComponent<OutlineEffect> = forwardRef(function Outline(
-  props: OutlineProps,
+  { 
+    selection = [], 
+    selectionLayer = 10, 
+    ...props
+}: OutlineProps,
   ref: Ref<OutlineEffect>
 ) {
   const { scene, camera } = useThree()
+
   const effect = useMemo(
-    () =>
-      new OutlineEffect(scene, camera, {
+    () =>{
```

Comment:
It looks like maybe prettier failed on commit here. I'll submit a PR shortly to convert to a more similar setup to the v5 branch in r3f